In [21]:
import pandas as pd
import numpy as np
from ast import literal_eval

***
Contents 기반 추천 해보기
***

In [77]:
df = pd.read_csv('wondoofin.csv')
cat_df = pd.read_csv('category.csv')

In [78]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 559 entries, 0 to 558
Data columns (total 26 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 559 non-null    int64 
 1   이름                 559 non-null    object
 2   로스터리               559 non-null    object
 3   타입                 559 non-null    object
 4   로스팅 포인트            559 non-null    object
 5   지속가능성              113 non-null    object
 6   컵 노트               559 non-null    object
 7   바디감                559 non-null    int64 
 8   신맛                 559 non-null    int64 
 9   단맛                 559 non-null    int64 
 10  쓴맛                 559 non-null    int64 
 11  커피 소개              559 non-null    object
 12  나라                 543 non-null    object
 13  로스터리 소개            559 non-null    object
 14  로스터리 주소            550 non-null    object
 15  이름.1               559 non-null    object
 16  식품의 유형             554 non-null    object
 1

In [91]:
df.columns

Index(['id', '이름', '로스터리', '타입', '로스팅 포인트', '지속가능성', '컵 노트', '바디감', '신맛', '단맛',
       '쓴맛', '커피 소개', '나라', '로스터리 소개', '로스터리 주소', '이름.1', '식품의 유형',
       '제조원 및 소재지', '유통기한', '제조일자', '내용량', '보관 방법', '원재료 및 함량',
       '제품문의 관련 주소 및 전화번호', '카페인', '로스터리ID'],
      dtype='object')

***
null 값 처리
- 지속가능성 : 113
- 나라 : 543
- 로스터리 주소 : 550
- 식품의 유형 : 554
- 제조원 및 소재지 : 557/559
- 유통기한 : 557
- 보관 방법 : 545
***

In [79]:
# 리스트 처리
df['컵 노트'] = df['컵 노트'].apply(literal_eval)
df['컵 노트'][0]

['토피', '맥아', '흑설탕']

***

In [80]:
#################################
########## 지속 가능성 ##########
#################################
for i in range(len(df)):
    if df['지속가능성'][i] is np.nan:
        df.at[i, '지속가능성'] = 0
        
df['지속가능성']

0         0
1         0
2       유기농
3      공정무역
4         0
       ... 
554       0
555       0
556       0
557       0
558       0
Name: 지속가능성, Length: 559, dtype: object

***

In [81]:
###########################
########## 나라 ##########
###########################
df['나라'] = df['나라'].str.split(', ')
df['나라'][0]

['브라질', '과테말라']

In [82]:
for i in range(len(df)):
    if df['나라'][i] is np.nan:
        df.at[i, '나라'] = 0
        
df['나라']

0           [브라질, 과테말라]
1                 [세라도]
2                 [아리차]
3             [브라질, 인도]
4              [리무 볼레소]
             ...       
554                   0
555       [에티오피아, 콜롬비아]
556         [브라질, 콜롬비아]
557               [파젠다]
558    [과테말라, 콜롬비아, 인도]
Name: 나라, Length: 559, dtype: object

***

In [83]:
print(df.loc[df['로스터리 주소'].isna(), 'id'].tolist())

[4701, 4692, 4613, 4614, 4695, 4694, 4612, 4611, 4693]


In [84]:
df.loc[df['로스터리 주소'].isna(), '로스터리 주소'] = 0

In [87]:
print(df.loc[df['제조원 및 소재지'].isna(), 'id'].tolist())

[274, 272]


In [93]:
print('로스터리 : ', df.loc[df['id'] == 4701, '로스터리'])
print('로스토리 주소 : ', df.loc[df['id'] == 4701, '로스터리 주소'])
print('제조원 : ', df.loc[df['id'] == 4701, '제조원 및 소재지'])

로스터리 :  17    커피네츄럴스페셜티
Name: 로스터리, dtype: object
로스토리 주소 :  17    0
Name: 로스터리 주소, dtype: object
제조원 :  17    경기도 고양시 덕양구 행신동 741-17 1층
Name: 제조원 및 소재지, dtype: object
